crates/swc_ecma_transforms_compat/src/es2015/classes/constructor.rs

```diff
@@ -337,41 +336,42 @@ impl Fold for ConstructorFolder<'_> {
                 }) => {
                     let old = self.cur_this_super.take();
 
-                    let mut e = e.fold_children_with(self);
+                    e.visit_mut_children_with(self);
 
                     let this_super = self.cur_this_super.take().unwrap();
                     self.cur_this_super = old;
 
                     e.args.insert(0, this_super.as_arg());
-
-                    CallExpr {
-                        span: e.span,
-                        callee: e
-                            .callee
+                    e.callee.map_with_mut(|callee| {
```

Comment:
this is q for @kdy1 : when do we recommend direct assignment vs. map_with_mut? I used direct assignment in most cases mostly not knowing map_with_mut.